ios/FluentUI/List/ListItem.swift

```diff
@@ -39,8 +46,12 @@ public struct ListItem<LeadingContent: View,
         self.leadingContent = leadingContent
         self.trailingContent = trailingContent
         self.action = action
+
         let layoutType = ListItem.layoutType(subtitle: subtitle, footer: footer)
-        self.tokenSet = ListItemTokenSet(customViewSize: { layoutType.leadingContentSize })
+        let leadingContentSize = layoutType.leadingContentSize
+        let state = ListItemStateImpl(leadingContentSize: leadingContentSize)
+        self.state = state
```

Comment:
This does match the existing pattern in other components, but it's one we'd like to move away from due to its complexity. Could we simply store `leadingContentSize` as a var on the `ListItem` itself and do away with `ListItemState` altogether?

Reply:
You won't be able to do that either, because a SwiftUI `View` has to be a `struct`. What if you just don't make it `@Published`? A mutating var on the struct should force a re-render automatically, if I'm remembering my SwiftUI correctly.

Reply:
ooh I didn't know that about views, turns out I don't know my swiftui very well hehe

Reply:
but I believe that doesn't work because the compiler is complaining `Escaping closure captures mutating 'self' parameter`

Reply:
spoke offline, going a different approach and following what `FluentButtonStyle` does instead